subsys/bluetooth/host/l2cap.c

```diff
@@ -539,11 +540,12 @@ static int l2cap_ecred_conn_req(struct bt_l2cap_chan **chan, int channels)
 	req->mtu = sys_cpu_to_le16(ch->rx.mtu);
 	req->mps = sys_cpu_to_le16(ch->rx.mps);
 	req->credits = sys_cpu_to_le16(ch->rx.init_credits);
+	req_psm = ch->psm;
 
 	for (i = 0; i < channels; i++) {
 		ch = BT_L2CAP_LE_CHAN(chan[i]);
 
-		__ASSERT(ch->psm == req->psm,
+		__ASSERT(ch->psm == req_psm,
```

Comment:
This may be a simpler and more obvious change, but either way works. 
```suggestion
		__ASSERT(sys_cpu_to_le16(ch->psm) == req->psm,
```

Reply:
> This may be a simpler and more obvious change, but either way works.

Sure, but it was to avoid adding the redundant byte swapping code.

Reply:
\- and I think it makes most sense to stay in "cpu" domain.